lib/api/CInferenceModelDefinition.cc

```diff
@@ -408,6 +533,62 @@ void CInferenceModelDefinition::dependentVariableColumnIndex(std::size_t depende
     m_DependentVariableColumnIndex = dependentVariableColumnIndex;
 }
 
+CInferenceModelDefinition::TSizeInfoUPtr CInferenceModelDefinition::sizeInfo() const {
+    return std::make_unique<CSizeInfo>(*this);
+}
+
+CInferenceModelDefinition::CSizeInfo::CSizeInfo(const CInferenceModelDefinition& definition)
+    : m_Definition{definition} {
+}
+
+std::string CInferenceModelDefinition::CSizeInfo::jsonString() {
+    std::ostringstream stream;
+    {
```

Comment:
I think it is worth a comment to say this scope block is to finish writing the object in the `CJsonOutputStreamWrapper` destructor.

Reply:
Although you remove them afterwards so is this necessary: maybe remove this and update `std::string resultString(jsonStr, 1, jsonStr.size() - 2)` to `std::string resultString(jsonStr, 1, jsonStr.size() - 1)`?

Reply:
I aded the comment, but if I remove the scope and change `-2` to `-1` I fail the tests.

Reply:
That seems weird to me since I thought all the destructor did was close the array, but maybe some detail of rapidjson, but happy to leave as is.

Reply:
A better option might be to add an optional argument to the `CJsonOutputStreamWrapper` constructor to say whether to output JSON or ND-JSON.

About 6-7 years ago JSON purists were in the ascendency and ND-JSON was considered impure.  So we got into the situation where our programs output multi-gigabyte JSON arrays just so that their entire output is a valid JSON document.  On the receiving side this has to be parsed using a stream parser rather than waiting for a complete JSON document, so the array structure is completely pointless.

Now ND-JSON is considered the sane way to deal with a stream of JSON documents, so we should use it for new types of output.

It isn't very hard to change `CJsonOutputStreamWrapper` to do this. In ND-JSON mode it would:

- Never write `JSON_ARRAY_START`
- Never write `JSON_ARRAY_END`
- Write `\n` wherever it currently writes `JSON_ARRAY_DELIMITER`

Then you don't need the inefficient string manipulation at the end of this method, and there's a side benefit that one of our core classes encourages the more modern approach for the future.

Reply:
This is a good point, I will do it in a separate PR though. See #1329